kernel/kernel-api/src/main/java/io/delta/kernel/internal/data/TransactionStateRow.java

```diff
@@ -39,21 +43,25 @@ public class TransactionStateRow extends GenericRow {
           .add(
               "configuration",
               new MapType(StringType.STRING, StringType.STRING, false /* valueContainsNull */))
-          .add("tablePath", StringType.STRING);
+          .add("tablePath", StringType.STRING)
+          .add("writerFeatures", new ArrayType(StringType.STRING, false /* containsNull */));
 
   private static final Map<String, Integer> COL_NAME_TO_ORDINAL =
       IntStream.range(0, SCHEMA.length())
           .boxed()
           .collect(toMap(i -> SCHEMA.at(i).getName(), i -> i));
 
   public static TransactionStateRow of(
-      Metadata metadata, String tablePath, StructType physicalSchema) {
+      Metadata metadata, Protocol protocol, String tablePath, StructType physicalSchema) {
     HashMap<Integer, Object> valueMap = new HashMap<>();
     valueMap.put(COL_NAME_TO_ORDINAL.get("logicalSchemaString"), metadata.getSchemaString());
     valueMap.put(COL_NAME_TO_ORDINAL.get("physicalSchemaString"), physicalSchema.toJson());
     valueMap.put(COL_NAME_TO_ORDINAL.get("partitionColumns"), metadata.getPartitionColumns());
     valueMap.put(COL_NAME_TO_ORDINAL.get("configuration"), metadata.getConfigurationMapValue());
     valueMap.put(COL_NAME_TO_ORDINAL.get("tablePath"), tablePath);
+    valueMap.put(
+        COL_NAME_TO_ORDINAL.get("writerFeatures"),
+        buildArrayValue(new ArrayList<>(protocol.getWriterFeatures()), StringType.STRING));
```

Comment:
given we just need to know whether cluster is supported or not (boolean), can we just pass that in the state than just the writerFeatures? It is odd to just pass writerFeatures from the protocol and nothing else.